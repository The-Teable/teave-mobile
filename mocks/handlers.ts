import { rest } from "msw";
import { UserWishProduct } from "../src/services/model/wishSchema";

type UserWishProductRequsetBody = {
  body: {
    tea_id: number;
  };
};

let wishProducts = [
  {
    id: 0,
    name: "카라멜 티메리카노 ",
    brand: "큐앤리브리즈",
    type: "",
    flavor: "",
    caffeine: "",
    efficacies: "",
    price: 15000,
    image_url: "/image/goods1.svg",
  },
  {
    id: 1,
    name: "카라멜 티메리카노 ",
    brand: "큐앤리브리즈",
    type: "",
    flavor: "",
    caffeine: "",
    efficacies: "",
    price: 15000,
    image_url: "/image/goods2.svg",
  },
  {
    id: 2,
    name: "카라멜 티메리카노 ",
    brand: "큐앤리브리즈",
    type: "",
    flavor: "",
    caffeine: "",
    efficacies: "",
    price: 15000,
    image_url: "/image/goods3.svg",
  },
  {
    id: 3,
    name: "카라멜 티메리카노 ",
    brand: "큐앤리브리즈",
    type: "",
    flavor: "",
    caffeine: "",
    efficacies: "",
    price: 15000,
    image_url: "/image/goods1.svg",
  },
];

export const handlers = [
  rest.get(
    process.env.NEXT_PUBLIC_LS_URL + "/user-wish-product/",
    (_req, res, ctx) => {
      return res(ctx.json<UserWishProduct[]>(wishProducts));
    }
  ),
  rest.post(
    process.env.NEXT_PUBLIC_LS_URL + "/user-wish-product/",
    (_req, res, ctx) => {
      // const { authToken } = req.cookies;
      return res(ctx.status(201));
    }
  ),
  rest.post(
    process.env.NEXT_PUBLIC_LS_URL + "/user-wish-product/delete",
    (req: UserWishProductRequsetBody, res, ctx) => {
      const { tea_id } = req.body;
      wishProducts = wishProducts.filter(({ id }) => tea_id !== id);
      return res(ctx.status(204));
    }
  ),
];
