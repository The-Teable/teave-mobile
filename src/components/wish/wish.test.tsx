import WishPage from "./page";
import { render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

describe("wish page", () => {
  if (process.env.NEXT_PUBLIC_API_MOCKING === "enabled") {
    require("../../mocks");
  }

  const queryClient = new QueryClient();
  it("matches snapshot", () => {
    const utils = render(
      <QueryClientProvider client={queryClient}>
        <WishPage />
      </QueryClientProvider>
    );
    expect(utils.container).toMatchSnapshot();
  });
});
