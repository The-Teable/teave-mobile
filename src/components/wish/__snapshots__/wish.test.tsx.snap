// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wish page matches snapshot 1`] = `
<div>
  <div
    class="CenteredContainer-sc-1yuzjpv-0 TitleHeader__Header-sc-11tys1g-0 fqZIID kzEIQl"
  >
    <div
      class="TitleHeader__Button-sc-11tys1g-1 TitleHeader__GoBackButton-sc-11tys1g-3 giDEjW iuJpAG"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <h1
      class="TitleHeader__Title-sc-11tys1g-2 ixFut"
    >
      찜한 상품
    </h1>
  </div>
  <div
    class="TitleHeader__Padding-sc-11tys1g-6 bmtQFo"
  />
  <div
    class="CenteredContainer-sc-1yuzjpv-0 fqZIID"
  />
  <div
    class="TabBar__Padding-sc-wdzu00-4 iUwhbr"
  />
  <nav
    class="CenteredContainer-sc-1yuzjpv-0 TabBar__Container-sc-wdzu00-0 fqZIID gVFLHI"
  >
    <div
      class="TabBar__Item-sc-wdzu00-1 hrEikW"
      href="/home"
    >
      <div
        class="TabBar__Icon-sc-wdzu00-2 dVOLca"
      />
      <p
        class="TabBar__Text-sc-wdzu00-3 hhdEZh"
      >
        홈
      </p>
    </div>
    <div
      class="TabBar__Item-sc-wdzu00-1 hrEikW"
      href="/search"
    >
      <div
        class="TabBar__Icon-sc-wdzu00-2 bPGhcH"
      />
      <p
        class="TabBar__Text-sc-wdzu00-3 hhdEZh"
      >
        검색
      </p>
    </div>
    <div
      class="TabBar__Item-sc-wdzu00-1 hrEikW"
      href="/wish"
    >
      <div
        class="TabBar__Icon-sc-wdzu00-2 jZLMVx"
      />
      <p
        class="TabBar__Text-sc-wdzu00-3 hhdEZh"
      >
        찜
      </p>
    </div>
    <div
      class="TabBar__Item-sc-wdzu00-1 hrEikW"
      href="/cart"
    >
      <div
        class="TabBar__Icon-sc-wdzu00-2 zuHol"
      />
      <p
        class="TabBar__Text-sc-wdzu00-3 hhdEZh"
      >
        장바구니
      </p>
    </div>
    <div
      class="TabBar__Item-sc-wdzu00-1 hrEikW"
      href="/mypage"
    >
      <div
        class="TabBar__Icon-sc-wdzu00-2 ksYoco"
      />
      <p
        class="TabBar__Text-sc-wdzu00-3 hhdEZh"
      >
        마이페이지
      </p>
    </div>
  </nav>
</div>
`;
